src/sonic_ax_impl/mibs/vendor/cisco/ciscoEntityFruControlMIB.py

```diff
@@ -18,41 +52,60 @@ def __init__(self):
         """
         init the handler
         """
-        self.psuutil = None
+        self.statedb = SonicV2Connector()
+        self.statedb.connect(self.statedb.STATE_DB)
 
-        try:
-            module = imp.load_source(PSU_PLUGIN_MODULE_NAME, PSU_PLUGIN_MODULE_PATH)
-        except ImportError as e:
-            mibs.logger.error("Failed to load PSU module '%s': %s" % (PSU_PLUGIN_MODULE_NAME, str(e)), True)
-            return
-        except FileNotFoundError as e:
-            mibs.logger.error("Failed to get platform specific PSU module '%s': %s" % (PSU_PLUGIN_MODULE_NAME, str(e)), True)
-            return
+    def _getPsuNum(self):
+        """
+        Get PSU number
+        :return: the number of supported PSU
+        """
+        chassis_name = CHASSIS_INFO_KEY_TEMPLATE.format(1)
+        chassis_info = self.statedb.get_all(self.statedb.STATE_DB, mibs.chassis_info_table(chassis_name))
+        psu_num = get_chassis_data(chassis_info)
 
-        try:
-            platform_psuutil_class = getattr(module, PSU_PLUGIN_CLASS_NAME)
-            self.psuutil = platform_psuutil_class()
-        except AttributeError as e:
-            mibs.logger.error("Failed to instantiate '%s' class: %s" % (PLATFORM_SPECIFIC_CLASS_NAME, str(e)), True)
+        return int(psu_num[0])
+
+    def _getPsuPresence(self, psu_index):
+        """
+        Get PSU presence
+        :return: the presence of particular PSU
+        """
+        psu_name = PSU_INFO_KEY_TEMPLATE.format(psu_index)
+        psu_info = self.statedb.get_all(self.statedb.STATE_DB, mibs.psu_info_table(psu_name))
+        presence, status = get_psu_data(psu_info)
+
+        return presence == PSU_PRESENCE_OK
+
+    def _getPsuStatus(self, psu_index):
+        """
+        Get PSU status
+        :return: the status of particular PSU
+        """
+        psu_name = PSU_INFO_KEY_TEMPLATE.format(psu_index)
+        psu_info = self.statedb.get_all(self.statedb.STATE_DB, mibs.psu_info_table(psu_name))
+        presence, status = get_psu_data(psu_info)
+
+        return status == PSU_STATUS_OK
 
     def _getPsuIndex(self, sub_id):
         """
         Get the PSU index from sub_id
         :return: the index of supported PSU
         """
-        if not self.psuutil or not sub_id or len(sub_id) > 1:
+        if not sub_id or len(sub_id) > 1:
             return None
 
         psu_index = int(sub_id[0])
 
         try:
-            num_psus = self.psuutil.get_num_psus()
+            psu_num = self._getPsuNum()
```

Comment:
Why change variable name (and new related getter function name) to "psu_num"? To me, "psu_num" infers the index of a given PSU. If you prefer to start the variable name with "psu_" (as opposed to "num_psus"), I suggest renaming to "psu_count".

Also, please follow PEP8 style rules for consistency (i.e., function names should not be camelCase, but rather lowercase with words separated by underscores as necessary.

Reply:
@jleveque The reason for that is official Spec name convention: https://github.com/Azure/SONiC/blob/master/doc/pmon/pmon-enhancement-design.md

1.5.2 Chassis Table
psu_num                 = INT                            ; psu numbers on the chassis

Reply:
I see. Would you consider changing the variable names?

Reply:
@jleveque sure. Will do it.

Reply:
@jleveque done.

Reply:
@nazariig: I see that you aligned the function name style with PEP8 guidelines, but the variable name is still named `psu_num` (as is the related getter function). Are you still planning to change it?

Reply:
@jleveque done. The variable names were aligned as well.

Reply:
Thanks. You decided on "num_psus" instead of something else like "psu_count"? If you want the variable name to start with "psu" I would recommend psu_count. We didn't need to go back to "num_psus", but I just think "psu_num" is confusing.

Reply:
@jleveque NP. I decided to stay with original name. Do you want me to force psu_count? I don't care to much regarding variables naming, but rather the stuff to be working. The idea of these changes to fix SNMP test.

Reply:
I'm fine either way. I just wanted to make it clear that I'm not set on the old name.

Reply:
@jleveque ok. Got it. Thanks for the clarifications.